res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+    // Light up FX quick effect chain selector buttons
+    midi.sendShortMsg(0x96+i, 0x64, 0x30);
+    midi.sendShortMsg(0x96+i, 0x65, 0x30);
+  }
+
+  DJCi500.tempoTimer = engine.beginTimer(250, DJCi500.tempoLEDs);
+
+  // FX buttons, light them to signal the current deck 1 and 2 as active
+  midi.sendShortMsg(0x90, 0x14, 0x7F);
+  midi.sendShortMsg(0x90, 0x15, 0x7F);
+
+  // Create the deck objects
+  DJCi500.deckA = new DJCi500.Deck([1, 3], 1);
+  DJCi500.deckB = new DJCi500.Deck([2, 4], 2);
+  DJCi500.deckA.setCurrentDeck("[Channel1]");
+  DJCi500.deckB.setCurrentDeck("[Channel2]");
+
+  // Update the fx rack selection
+  DJCi500.fxSelIndicator(0, "[EffectRack1_EffectUnit1]", 0, 0);
+  DJCi500.fxSelIndicator(0, "[EffectRack1_EffectUnit2]", 0, 0);
+
+  DJCi500.fxEnabledIndicator(0, "[QuickEffectRack1_[Channel1]]", 0, 0);
+  DJCi500.fxEnabledIndicator(0, "[QuickEffectRack1_[Channel2]]", 0, 0);
+};
+
+// Crossfader control, set the curve
+DJCi500.crossfaderSetCurve = function(channel, control, value, _status, _group) {
+  switch(value) {
+    case 0x00:
+      // Mix
+      script.crossfaderCurve(0,0,127);
+      DJCi500.xFaderScratch = false;
+      break;
+    case 0x7F:
+      // Scratch
+      script.crossfaderCurve(127,0,127);
+      DJCi500.xFaderScratch = true;
+      break;
+  }
+}
+
+// Crossfader enable or disable
+DJCi500.crossfaderEnable = function(channel, control, value, _status, _group) {
+  if(value) {
+    DJCi500.crossfaderEnabled = true;
+  } else {
+    DJCi500.crossfaderEnabled = false;
+    engine.setValue("[Master]", "crossfader", 0);    // Set the crossfader in the middle
+  }
+}
+
+// Crossfader function
+DJCi500.crossfader = function(channel, control, value, status, group) {
+  if (DJCi500.crossfaderEnabled) {
+    // Eventine's crossfader scratch mode
+    if (DJCi500.xFaderScratch) {
+      var result = 0;
+      if (value <= 0) {
+        result = -1;
+      }
+      else if (value >= 127) {
+        result = 1;
+      }
+      else {
+        result = Math.tan((value-64)*Math.PI/2/63)/32;
+      }
+      engine.setValue(group, "crossfader", result);
+    }
+    else {
+      engine.setValue(group, "crossfader", (value/64)-1);
+    }
+  }
+}
+
+// Browser button. We move it to a custom JS function to avoid having to focus the Mixxx window for it to respond
+DJCi500.moveLibrary = function(channel, control, value, status, group) {
+  if (value > 0x3F) {
+    if (DJCi500.browserOffFocusMode) {
+      engine.setValue('[Playlist]', 'SelectTrackKnob', -1);
+    } else {
+      engine.setValue('[Library]', 'MoveUp', 1);
+    }
+  } else {
+    if (DJCi500.browserOffFocusMode) {
+      engine.setValue('[Playlist]', 'SelectTrackKnob', 1);
+    } else {
+      engine.setValue('[Library]', 'MoveDown', 1);
+    }
+  }
+}
+
+DJCi500.spinback_button = function(channel, control, value, status, group) {
+  var deck = parseInt(group.substring(8,9)); // work out which deck we are using

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
    var deck = script.deckFromGroup(group);      
```